assets/pages/new-contributions-landing/components/StripePaymentRequestButton.jsx

```diff
@@ -0,0 +1,242 @@
+  // until the async canMakePayment() function has been called on the paymentRequest object.
+  if (!props.paymentRequest) {
+    initialisePaymentRequest({
+      stripe: props.stripe,
+      country: props.country,
+      currency: props.currency,
+      amount,
+      setCanMakeApplePayPayment: props.setCanMakeApplePayPayment,
+      setPaymentRequest: props.setPaymentRequest,
+      onPaymentAuthorised: props.onPaymentAuthorised,
+      isTestUser: props.isTestUser,
+      updateEmail: props.updateEmail,
+    });
+    return null;
+  }
+
+  if (!Number.isNaN(amount) && props.paymentRequest) {
+    props.paymentRequest.update({
+      total: {
+        label: 'The Guardian',
+        amount: amount * 100,
+      },
+    });
+  }
+
+  return (props.canMakeApplePayPayment === true) ? (
+    <div className="stripe-payment-request-button__container">
+      <PaymentRequestButtonElement
+        paymentRequest={props.paymentRequest}
+        className="stripe-payment-request-button__button"
+        style={{
+          paymentRequestButton: {
+            theme: 'dark',
+            height: '42px',
+          },
+        }}
+        onClick={
+          (event) => {
+            event.preventDefault();
+            props.setStripePaymentRequestButtonClicked();
+            const amountIsValid =
+              checkAmountOrOtherAmount(
+                props.selectedAmounts,
+                props.otherAmounts,
+                props.contributionType,
+                props.countryGroupId,
+              );
+            if (props.paymentRequest && amountIsValid) {
+              props.paymentRequest.show();
+            }
+          }
+        }
+      />
+      <div className="stripe-payment-request-button__divider">
+        or
+      </div>
+    </div>
+  ) : null;
+}
+
+
+// ----- Component ----- //
+
+function StripePaymentRequestButton(props: PropTypes) {
+
+  const showApplePay = props.stripeCheckout
+    && isInStripePaymentRequestAllowedCountries(props.country)
+    && props.currency !== 'AUD'
+    && props.applePayTestVariant === 'applePay';
+
+  // TODO: set up for AU
+  if (showApplePay) {
+    const key = getStripeKey('ONE_OFF', props.currency, props.isTestUser);
+
+    return (
+      <div className={hiddenIf(props.contributionType !== 'ONE_OFF', 'stripe-payment-request-button')}>
+        <StripeProvider apiKey={key}>
+          <Elements>
+            <PaymentRequestButton
+              country={props.country}
+              currency={props.currency.toLowerCase()}
+              isTestUser={props.isTestUser}
+              contributionType={props.contributionType}
+            />
+          </Elements>
+        </StripeProvider>
+      </div>
+    );
+  }
+  return null;
+}
+
+// ----- Auxiliary components ----- //
+
+const PaymentRequestButton = injectStripe(connect(mapStateToProps, mapDispatchToProps)(paymentRequestButton));
+
+// ----- Default props----- //
+
+export default StripePaymentRequestButton;
```

Comment:
this file is a bit of a monster, just wonder is there any way of splitting either the file or even the main `function paymentRequestButton` down at all?

Reply:
Done